### PR TITLE
Clang format linter

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-925-502fbb8326c67ccd9520d52e1327b590bf16aabd-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-925-502fbb8326c67ccd9520d52e1327b590bf16aabd-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-925-502fbb8326c67ccd9520d52e1327b590bf16aabd-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-925-502fbb8326c67ccd9520d52e1327b590bf16aabd-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-925-b5e13f0d03d93cf41fc571182eecbb142e508893-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-925-b5e13f0d03d93cf41fc571182eecbb142e508893-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-925-b5e13f0d03d93cf41fc571182eecbb142e508893-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-925-b5e13f0d03d93cf41fc571182eecbb142e508893-x86_64.apk)

**Comment by 3nids** [comment]
nice ;)

**Comment by 3nids** [comment]
have you tried to set `AccessModifierOffset` to 0 or +2 ?

**Comment by m-kuhn** [comment]
That's what `AccessModifierOffset: 2` does:

```
class QgsGeometryWrapper : public QObject
{
  Q_OBJECT
  Q_PROPERTY( QgsGeometry qgsGeometry READ qgsGeometry WRITE setQgsGeometry NOTIFY qgsGeometryChanged )
  Q_PROPERTY( QgsCoordinateReferenceSystem crs READ crs WRITE setCrs NOTIFY crsChanged )

    public:
  explicit QgsGeometryWrapper( QObject *parent = nullptr )
    : QObject( parent )
  {}

```

File: .clang-format [hunk 1]
```diff
@@ -54,14 +55,21 @@ FixNamespaceComments: true
...
+    Priority:        400
+  - Regex:           '^".*'
+    Priority:        100
+  - Regex:           '.*'
+    Priority:        1
+IncludeIsMainRegex: false
+IncludeBlocks: Regroup
 IndentCaseLabels: true
-IndentPPDirectives: false
 IndentWidth: 2
```
Comment (3nids) [inline .clang-format:73]:
```suggestion
IndentWidth: 4
```

looking at the issue here: https://bugs.llvm.org/show_bug.cgi?id=19056
I think we were better using a 4 indent, so there would be no modification of the indent in the headers (by keeping the access modifiers. offset to -2)

Comment (m-kuhn) [reply .clang-format:73]: Won't that have an immense impact on all code in cpp files?
What we'd want is only the `class` and `struct` set to 4.

Comment (3nids) [reply .clang-format:73]: indeed

**Comment by 3nids** [comment]
@m-kuhn do you want to move on now with this one?

**Pull request closed without merging**